helm/kanister-operator/values.yaml

```diff
@@ -36,6 +36,20 @@ controller:
     # false : kanister-prometheus framework has been disabled
     # true: kanister-prometheus framework has been enabled
     enabled: false
+  # Pod-level security context settings
+  podSecurityContext:
+  # runAsUser: 1000
+  # fsGroup: 2000
+  # runAsNonRoot: true
```

Comment:
I think we should remove these comments. Same for `containerSecurityContext`.

Reply:
@viveksinghggits if i do this. Then, i think there is not need to test for `Only Container security context is set ` and `Only pod security context is set.

Reply:
sorry, I didn't understand this. By removing the comments are are not actually changing anything right? because it's (the helm fields) already commented out. So it it's commented out just removing them should not make any difference.

Reply:
The second test case throws the following error when running `make helm-test`:

```
Test name: Only Container security context is set
helm_test.go:284:
    c.Assert(obj.Spec.Template.Spec.SecurityContext, check.DeepEquals, tc.expectedPodSecurity)
... obtained *v1.PodSecurityContext = &v1.PodSecurityContext{SELinuxOptions:(*v1.SELinuxOptions)(nil), WindowsOptions:(*v1.WindowsSecurityContextOptions)(nil), RunAsUser:(*int64)(0xc0002b5958), RunAsGroup:(*int64)(nil), RunAsNonRoot:(*bool)(0xc0002b59a0), SupplementalGroups:[]int64(nil), FSGroup:(*int64)(0xc0002b5988), Sysctls:[]v1.Sysctl(nil), FSGroupChangePolicy:(*v1.PodFSGroupChangePolicy)(nil), SeccompProfile:(*v1.SeccompProfile)(nil)} ("&PodSecurityContext{SELinuxOptions:nil,RunAsUser:*1000,RunAsNonRoot:*true,SupplementalGroups:[],FSGroup:*2000,RunAsGroup:nil,Sysctls:[]Sysctl{},WindowsOptions:nil,FSGroupChangePolicy:nil,SeccompProfile:nil,}")
... expected *v1.PodSecurityContext = (*v1.PodSecurityContext)(nil) ("nil")

FAIL: helm_test.go:203: HelmTestSuite.TestSecurityContextInHelmChart
```

- The `expectedPodSecurity` in this test case is `nil`, which works fine when I comment out the `podSecurityContext` in `values.yaml`  because the values for `podsecuritycontext` becomes`nil` and match the expected value.
- However, once the `podSecurityContext` is uncommented, the  value for `podsecuritycontext` changes, but the test case still expects `nil`. 
- If I modify the expected value to match what's in `values.yaml`, it ends up being the same as the first test case.
- Similar for third test case.

Reply:
hmm, I think we we are doing something wrong. do you agree that removing the commented fields from `values.yaml` should not have any affect anywhere?

Reply:
yes , it affect . The value that we have taken in `test file` would take precendence over the value that are present in` value.yaml`. if the values on our test file is nil or empty then values will be taken from `values.yaml`.

Reply:
> over the value that are present in value.yaml

my understanding is we are not configuring any values in values.yaml right? because we have just commented them out.

Reply:
yes, we are not using whenever it is commented.

Reply:
If we actually want to run operator  with security context , then we have to uncomment the changes. And as I told earlier test 2 and test 3 is no more needed because we have enabled both security context.

Reply:
Any suggestion you would add here ? Thank you

Reply:
> Any suggestion you would like to add here ? Thank you